service/src/main/resources/api/service_openapi.yaml

```diff
@@ -1965,7 +2010,26 @@ components:
           $ref: '#/components/schemas/JobReport'
         errorReport:
           $ref: '#/components/schemas/ErrorReport'
-
+    CloneReferencedResourceResult:
+      description: >-
+        API result class for cloning a referenced resource. Includes source workspace
+        and resource IDs for provenance, and uses the union-typed ResourceDescription object
+        for the cloned referenced resource details. If the effective cloning instructions are not
+        COPY_REFERENCE, then no clone is created and the resource is null.
+      type: object
+      properties:
+        effectiveCloningInstructions:
+          $ref: '#/components/schemas/CloningInstructionsEnum'
+        sourceWorkspaceId:
+          description: ID of the workspace of the source resource for this clone
+          type: string
+          format: uuid
+        sourceResourceId:
+          description: ID of the source resource
+          type: string
+          format: uuid
+        resource:
+          $ref: '#/components/schemas/ResourceDescription'
```

Comment:
I have mixed feelings about this. I would like to minimize the use of the union form. It is annoying to clients, as we have heard. I think we could put the clone operation at the per-resource level with a specific response and avoid it.

Reply:
I'll split them out; it won't be too bad.